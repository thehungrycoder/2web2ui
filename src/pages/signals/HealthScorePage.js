/* eslint-disable max-lines */
import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import { Panel, Grid } from '@sparkpost/matchbox';
import Page from './components/SignalsPage';
import BarChart from './components/charts/barchart/BarChart';
import DivergingBar from './components/charts/divergingBar/DivergingBar';
import HealthScoreActions from './components/actionContent/HealthScoreActions';
import TooltipMetric from './components/charts/tooltip/TooltipMetric';
import DateFilter from './components/filters/DateFilter';
import { HEALTH_SCORE_INFO, HEALTH_SCORE_COMPONENT_INFO, INJECTIONS_INFO, HEALTH_SCORE_COMPONENTS } from './constants/info';
import withHealthScoreDetails from './containers/HealthScoreDetailsContainer';
import { Loading } from 'src/components';
import Callout from 'src/components/callout';
import OtherChartsHeader from './components/OtherChartsHeader';
import ChartHeader from './components/ChartHeader';
import { formatFullNumber, roundToPlaces, formatNumber } from 'src/helpers/units';
import moment from 'moment';
import _ from 'lodash';

import SpamTrapsPreview from './components/previews/SpamTrapsPreview';
import EngagementRecencyPreview from './components/previews/EngagementRecencyPreview';
import styles from './DetailsPages.module.scss';

export class HealthScorePage extends Component {
  state = {
    selectedDate: null,
    selectedComponent: null
  }

  componentDidMount() {
    const { selected } = this.props;

    if (selected) {
      this.setState({ selectedDate: selected });
    }
  }

  componentDidUpdate(prevProps) {
    const { data } = this.props;
    const { selectedDate } = this.state;

    const dataSetChanged = prevProps.data !== data;
    let selectedDataByDay = _.find(data, ['date', selectedDate]);

    // Select last date in time series
    if (dataSetChanged && !selectedDataByDay) {
      selectedDataByDay = _.last(data);
      this.setState({ selectedDate: selectedDataByDay.date });
    }

    // Select first component weight type
    if (dataSetChanged) {
      const firstComponentType = _.get(selectedDataByDay, 'weights[0].weight_type');
      this.setState({ selectedComponent: firstComponentType });
    }
  }

  handleDateSelect = (node) => {
    this.setState({ selectedDate: _.get(node, 'payload.date') });
  }

  handleComponentSelect = (node) => {
    this.setState({ selectedComponent: _.get(node, 'payload.weight_type') });
  }

  getXAxisProps = () => {
    const { xTicks } = this.props;
    return {
      ticks: xTicks,
      tickFormatter: (tick) => moment(tick).format('M/D')
    };
  }

  renderContent = () => {
    const { data = [], loading, gap, empty, error } = this.props;
    const { selectedDate, selectedComponent } = this.state;

    const selectedWeights = _.get(_.find(data, ['date', selectedDate]), 'weights', []);
    const selectedWeightsAreEmpty = selectedWeights.every(({ weight }) => weight === null);
    const dataForSelectedWeight = data.map(({ date, weights }) => ({ date, ..._.find(weights, ['weight_type', selectedComponent]) }));

    let panelContent;

    if (empty) {
      panelContent = <Callout title='No Data Available'>Insufficient data to populate this chart</Callout>;
    }

    if (error) {
      panelContent = <Callout title='Unable to Load Data'>{error.message}</Callout>;
    }

    if (loading) {
      panelContent = (
        <div style={{ height: '220px', position: 'relative' }}>
          <Loading />
        </div>
      );
    }

    return (
      <Grid>
        <Grid.Column sm={12} md={7}>
          <Panel sectioned>
            <ChartHeader title='Health Score' tooltipContent={HEALTH_SCORE_INFO} />
            {panelContent || (
              <Fragment>
                <BarChart
                  gap={gap}
                  onClick={this.handleDateSelect}
                  selected={selectedDate}
                  timeSeries={data}
                  tooltipContent={({ payload = {}}) => (
                    <TooltipMetric label='Health Score' value={`${roundToPlaces(payload.health_score * 100, 1)}`} />
                  )}
                  yKey='health_score'
                  yAxisProps={{
                    tickFormatter: (tick) => tick * 100
                  }}
                  xAxisProps={this.getXAxisProps()}
                />
                <ChartHeader title='Injections' tooltipContent={INJECTIONS_INFO} />
                <BarChart
                  gap={gap}
                  height={190}
                  onClick={this.handleDateSelect}
                  selected={selectedDate}
                  timeSeries={data}
                  tooltipContent={({ payload = {}}) => (
                    <TooltipMetric label='Injections' value={formatFullNumber(payload.injections)} />
                  )}
                  yKey='injections'
                  yAxisProps={{
                    tickFormatter: (tick) => formatNumber(tick)
                  }}
                  xAxisProps={this.getXAxisProps()}
                />
                {(selectedComponent && !selectedWeightsAreEmpty) && (
                  <Fragment>
                    <ChartHeader title={HEALTH_SCORE_COMPONENTS[selectedComponent].chartTitle} />
                    <BarChart
                      gap={gap}
                      height={190}
                      onClick={this.handleDateSelect}
                      selected={selectedDate}
                      timeSeries={dataForSelectedWeight}
                      tooltipContent={({ payload = {}}) => (
                        <TooltipMetric label={selectedComponent} value={`${roundToPlaces(payload.weight_value * 100, 4)}%`} />
                      )}
                      yKey='weight_value'
                      yAxisProps={{
                        tickFormatter: (tick) => `${roundToPlaces(tick * 100, 3)}%`
                      }}
                      xAxisProps={this.getXAxisProps()}
                    />
                  </Fragment>
                )}
              </Fragment>
            )}
          </Panel>
        </Grid.Column>
        <Grid.Column sm={12} md={5} mdOffset={0}>
          <div className={styles.OffsetCol}>
            <ChartHeader
              title='Health Score Components'
              date={selectedDate}
              hideLine
              padding='1rem 0 1rem'
              tooltipContent={HEALTH_SCORE_COMPONENT_INFO}
            />
            {(!loading && selectedWeightsAreEmpty) && (
              <Callout>Insufficient data to populate this chart</Callout>
            )}
            {(!panelContent && !selectedWeightsAreEmpty) && (
              <DivergingBar
                data={selectedWeights}
                xKey='weight'
                yKey='weight_type'
                yLabel={({ value }) => _.get(HEALTH_SCORE_COMPONENTS[value], 'label')}
                tooltipContent={({ payload = {}}) => _.get(HEALTH_SCORE_COMPONENTS[payload.weight_type], 'info')}
                onClick={this.handleComponentSelect}
                selected={selectedComponent}
              />
            )}
            {!panelContent && <HealthScoreActions weights={selectedWeights} date={selectedDate} />}
          </div>
        </Grid.Column>
      </Grid>
    );
  }

  render() {
    const { facet, facetId } = this.props;

    return (
      <Page
        breadcrumbAction={{ content: 'Back to Overview', to: '/signals', component: Link }}
        dimensionPrefix='Health Score for'
        facet={facet}
        facetId={facetId}
        primaryArea={<DateFilter />}>
        {this.renderContent()}
        <OtherChartsHeader facet={facet} facetId={facetId} />
        <Grid>
          <Grid.Column xs={12} sm={6}>
            <SpamTrapsPreview />
          </Grid.Column>
          <Grid.Column xs={12} sm={6}>
            <EngagementRecencyPreview />
          </Grid.Column>
        </Grid>
      </Page>
    );
  }
}

export default withHealthScoreDetails(HealthScorePage);
