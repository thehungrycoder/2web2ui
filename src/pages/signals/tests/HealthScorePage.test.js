import { shallow } from 'enzyme';
import React from 'react';
import { HealthScorePage } from '../HealthScorePage';

describe('Signals Health Score Page', () => {
  let wrapper;
  let props;
  const data = [
    {
      date: '2017-01-01',
      weights: [
        { weight_type: 'Hard Bounces', weight: 0.5, weight_value: 0.25 },
        { weight_type: 'Complaints', weight: -0.5, weight_value: 0.25 }
      ]
    },
    {
      date: '2017-01-02',
      weights: [
        { weight_type: 'List Quality', weight: 0.8, weight_value: 0.25 },
        { weight_type: 'Other bounces', weight: -0.8, weight_value: 0.25 }
      ]
    }
  ];

  beforeEach(() => {
    props = {
      facetId: 'test.com',
      facet: 'sending-domain',
      data: [],
      selected: '2017-01-01',
      gap: 0.25,
      loading: false,
      empty: false,
      xTicks: []
    };
    wrapper = shallow(<HealthScorePage {...props}/>);
    wrapper.setProps({ data });
  });

  it('renders correctly when recieving data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders loading correctly', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper.find('Panel')).toMatchSnapshot();
  });

  it('renders empty correctly', () => {
    wrapper.setProps({ empty: true });
    expect(wrapper.find('Panel')).toMatchSnapshot();
  });

  it('renders error correctly', () => {
    wrapper.setProps({ error: { message: 'error message' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty weights', () => {
    wrapper.setProps({ data: [{ date: '2017-01-01' }]});
    expect(wrapper).toMatchSnapshot();
  });

  describe('local state', () => {
    it('handles date select', () => {
      wrapper.find('BarChart').at(0).simulate('click', { payload: { date: '2017-01-02' }});
      expect(wrapper.find('BarChart').at(0).prop('selected')).toEqual('2017-01-02');
      expect(wrapper.find('BarChart').at(1).prop('selected')).toEqual('2017-01-02');
      expect(wrapper.find('ChartHeader').at(3).prop('date')).toEqual('2017-01-02');
      expect(wrapper.find('HealthScoreActions')).toMatchSnapshot();
    });

    it('sets selected date on mount if provided one', () => {
      wrapper = shallow(<HealthScorePage {...props} selected='initial-selected'/>);
      expect(wrapper.find('BarChart').at(0).prop('selected')).toEqual('initial-selected');
      expect(wrapper.find('HealthScoreActions').prop('date')).toEqual('initial-selected');
    });

    it('uses last selected date if selected date is not in data', () => {
      wrapper = shallow(<HealthScorePage {...props} loading={true} selected='initial-selected'/>);
      wrapper.setProps({ data: [1, { date: 'last-date' }], loading: false });
      expect(wrapper.find('BarChart').at(0).prop('selected')).toEqual('last-date');
      expect(wrapper.find('HealthScoreActions').prop('date')).toEqual('last-date');
    });

    it('does not use last selected date if selected date is in data', () => {
      wrapper = shallow(<HealthScorePage {...props} selected='first-date'/>);
      wrapper.setProps({ data: [{ date: 'first-date' }, { date: 'last-date' }]});
      expect(wrapper.find('BarChart').at(0).prop('selected')).toEqual('first-date');
      expect(wrapper.find('HealthScoreActions').prop('date')).toEqual('first-date');
    });

    it('handles component select', () => {
      wrapper.find('DivergingBar').at(0).simulate('click', { payload: { weight_type: 'Complaints' }});
      expect(wrapper.find('DivergingBar').at(0).prop('selected')).toEqual('Complaints');
    });

    it('uses first component weight with an existing date and new data', () => {
      wrapper = shallow(<HealthScorePage {...props} selected='first-date'/>);
      wrapper.setProps({ data: [{ date: 'first-date', weights: [
        { weight_type: 'Hard Bounces', weight: 0.5, weight_value: 0.25 },
        { weight_type: 'Complaints', weight: 0.5, weight_value: 0.25 }
      ]}, { date: 'last-date' }]});
      expect(wrapper.find('DivergingBar').at(0).prop('selected')).toEqual('Hard Bounces');
    });

    it('uses first component weight of last date without an existing date and new data', () => {
      wrapper = shallow(<HealthScorePage {...props} selected='initial-date'/>);
      wrapper.setProps({ data: [{ date: 'first-date' }, { date: 'last-date', weights: [
        { weight_type: 'Hard Bounces', weight: 0.5, weight_value: 0.25 },
        { weight_type: 'Complaints', weight: 0.5, weight_value: 0.25 }
      ]}]});
      expect(wrapper.find('DivergingBar').at(0).prop('selected')).toEqual('Hard Bounces');
    });
  });

  describe('bar chart props', () => {
    it('renders tooltip content for health score', () => {
      const Tooltip = wrapper.find('BarChart').at(0).prop('tooltipContent');
      expect(shallow(<Tooltip payload={{ health_score: 0.12 }} />)).toMatchSnapshot();
    });

    it('renders tooltip content for injections', () => {
      const Tooltip = wrapper.find('BarChart').at(1).prop('tooltipContent');
      expect(shallow(<Tooltip payload={{ injections: 1000 }} />)).toMatchSnapshot();
    });

    it('renders tooltip content for selected component', () => {
      wrapper.find('DivergingBar').at(0).simulate('click', { payload: { weight_type: 'Complaints' }});
      const Tooltip = wrapper.find('BarChart').at(2).prop('tooltipContent');
      expect(shallow(<Tooltip payload={{ weight_value: 0.0012345 }} />)).toMatchSnapshot();
    });

    it('renders tooltip content for component weights', () => {
      const Tooltip = wrapper.find('DivergingBar').prop('tooltipContent');
      expect(shallow(<Tooltip payload={{ weight_type: 'Other bounces' }} />)).toMatchSnapshot();
    });

    it('renders y label content for component weights', () => {
      const label = wrapper.find('DivergingBar').prop('yLabel');
      expect(label({ value: 'eng cohorts: new, 14-day' })).toMatchSnapshot();
    });

    it('gets x axis props', () => {
      const axisProps = wrapper.find('BarChart').at(0).prop('xAxisProps');
      expect(axisProps).toMatchSnapshot();
      expect(axisProps.tickFormatter('2018-12-05')).toEqual('12/5');
    });

    it('renders health score y ticks', () => {
      const axisProps = wrapper.find('BarChart').at(0).prop('yAxisProps');
      expect(axisProps.tickFormatter(0.2468)).toEqual(24.68);
    });

    it('renders injections y ticks', () => {
      const axisProps = wrapper.find('BarChart').at(1).prop('yAxisProps');
      expect(axisProps.tickFormatter(2468)).toEqual('2.47K');
    });

    it('renders component y ticks', () => {
      const axisProps = wrapper.find('BarChart').at(2).prop('yAxisProps');
      expect(axisProps.tickFormatter(0.003)).toEqual('0.3%');
    });
  });
});
