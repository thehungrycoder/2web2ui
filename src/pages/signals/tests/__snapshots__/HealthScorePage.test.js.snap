// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signals Health Score Page bar chart props gets x axis props 1`] = `
Object {
  "tickFormatter": [Function],
  "ticks": Array [],
}
`;

exports[`Signals Health Score Page bar chart props renders tooltip content for component weights 1`] = `
"
      Other Bounces are defined as the share of attempted email injections that bounce for a variety
      of miscellaneous technical reasons.
    "
`;

exports[`Signals Health Score Page bar chart props renders tooltip content for health score 1`] = `
<TooltipMetric
  label="Health Score"
  value="12"
/>
`;

exports[`Signals Health Score Page bar chart props renders tooltip content for injections 1`] = `
<TooltipMetric
  label="Injections"
  value="1,000"
/>
`;

exports[`Signals Health Score Page bar chart props renders tooltip content for selected component 1`] = `
<TooltipMetric
  label="Complaints"
  value="0.1235%"
/>
`;

exports[`Signals Health Score Page bar chart props renders y label content for component weights 1`] = `"Engaged Recipients"`;

exports[`Signals Health Score Page local state handles date select 1`] = `
<HealthScoreActions
  date="2017-01-02"
  weights={
    Array [
      Object {
        "weight": 0.8,
        "weight_type": "List Quality",
        "weight_value": 0.25,
      },
      Object {
        "weight": -0.8,
        "weight_type": "Other bounces",
        "weight_value": 0.25,
      },
    ]
  }
/>
`;

exports[`Signals Health Score Page renders correctly when recieving data 1`] = `
<SignalsPage
  breadcrumbAction={
    Object {
      "component": [Function],
      "content": "Back to Overview",
      "to": "/signals",
    }
  }
  dimensionPrefix="Health Score for"
  facet="sending-domain"
  facetId="test.com"
  primaryArea={<Connect(DateFilter) />}
>
  <Grid>
    <Grid.Column
      md={7}
      sm={12}
    >
      <Panel
        sectioned={true}
      >
        <ChartHeader
          title="Health Score"
          tooltipContent="
  This is a predictive score that monitors your email health to identify problems before
  they negatively impact email delivery. This score is informed by message events,
  including bounces, spam trap hits, and engagement across our entire network.
"
        />
        <BarChart
          fill="#B157CE"
          gap={0.25}
          height={250}
          margin={
            Object {
              "bottom": 5,
              "left": 18,
              "right": 0,
              "top": 12,
            }
          }
          onClick={[Function]}
          selected="2017-01-01"
          timeSeries={
            Array [
              Object {
                "date": "2017-01-01",
                "weights": Array [
                  Object {
                    "weight": 0.5,
                    "weight_type": "Hard Bounces",
                    "weight_value": 0.25,
                  },
                  Object {
                    "weight": -0.5,
                    "weight_type": "Complaints",
                    "weight_value": 0.25,
                  },
                ],
              },
              Object {
                "date": "2017-01-02",
                "weights": Array [
                  Object {
                    "weight": 0.8,
                    "weight_type": "List Quality",
                    "weight_value": 0.25,
                  },
                  Object {
                    "weight": -0.8,
                    "weight_type": "Other bounces",
                    "weight_value": 0.25,
                  },
                ],
              },
            ]
          }
          tooltipContent={[Function]}
          width="99%"
          xAxisProps={
            Object {
              "tickFormatter": [Function],
              "ticks": Array [],
            }
          }
          xKey="date"
          yAxisProps={
            Object {
              "tickFormatter": [Function],
            }
          }
          yKey="health_score"
          yRange={
            Array [
              "auto",
              "auto",
            ]
          }
        />
        <ChartHeader
          title="Injections"
          tooltipContent="
  Injections are defined as the total number of messages successfully generated or relayed to SparkPost for delivery.
"
        />
        <BarChart
          fill="#B157CE"
          gap={0.25}
          height={190}
          margin={
            Object {
              "bottom": 5,
              "left": 18,
              "right": 0,
              "top": 12,
            }
          }
          onClick={[Function]}
          selected="2017-01-01"
          timeSeries={
            Array [
              Object {
                "date": "2017-01-01",
                "weights": Array [
                  Object {
                    "weight": 0.5,
                    "weight_type": "Hard Bounces",
                    "weight_value": 0.25,
                  },
                  Object {
                    "weight": -0.5,
                    "weight_type": "Complaints",
                    "weight_value": 0.25,
                  },
                ],
              },
              Object {
                "date": "2017-01-02",
                "weights": Array [
                  Object {
                    "weight": 0.8,
                    "weight_type": "List Quality",
                    "weight_value": 0.25,
                  },
                  Object {
                    "weight": -0.8,
                    "weight_type": "Other bounces",
                    "weight_value": 0.25,
                  },
                ],
              },
            ]
          }
          tooltipContent={[Function]}
          width="99%"
          xAxisProps={
            Object {
              "tickFormatter": [Function],
              "ticks": Array [],
            }
          }
          xKey="date"
          yAxisProps={
            Object {
              "tickFormatter": [Function],
            }
          }
          yKey="injections"
          yRange={
            Array [
              "auto",
              "auto",
            ]
          }
        />
        <ChartHeader
          title="Hard Bounces Percent"
        />
        <BarChart
          fill="#B157CE"
          gap={0.25}
          height={190}
          margin={
            Object {
              "bottom": 5,
              "left": 18,
              "right": 0,
              "top": 12,
            }
          }
          onClick={[Function]}
          selected="2017-01-01"
          timeSeries={
            Array [
              Object {
                "date": "2017-01-01",
                "weight": 0.5,
                "weight_type": "Hard Bounces",
                "weight_value": 0.25,
              },
              Object {
                "date": "2017-01-02",
              },
            ]
          }
          tooltipContent={[Function]}
          width="99%"
          xAxisProps={
            Object {
              "tickFormatter": [Function],
              "ticks": Array [],
            }
          }
          xKey="date"
          yAxisProps={
            Object {
              "tickFormatter": [Function],
            }
          }
          yKey="weight_value"
          yRange={
            Array [
              "auto",
              "auto",
            ]
          }
        />
      </Panel>
    </Grid.Column>
    <Grid.Column
      md={5}
      mdOffset={0}
      sm={12}
    >
      <div
        className="OffsetCol"
      >
        <ChartHeader
          date="2017-01-01"
          hideLine={true}
          padding="1rem 0 1rem"
          title="Health Score Components"
          tooltipContent="
  Your email Health Score is comprised of several signals, including bounces, spam trap hits, user
  engagement, and others.
"
        />
        <DivergingBar
          barHeight={35}
          data={
            Array [
              Object {
                "weight": 0.5,
                "weight_type": "Hard Bounces",
                "weight_value": 0.25,
              },
              Object {
                "weight": -0.5,
                "weight_type": "Complaints",
                "weight_value": 0.25,
              },
            ]
          }
          negativeFill="#DB2F2D"
          onClick={[Function]}
          positiveFill="#8CBE3C"
          selected="Hard Bounces"
          tooltipContent={[Function]}
          width="99%"
          xDomain={
            Array [
              "dataMin",
              "dataMax",
            ]
          }
          xKey="weight"
          yKey="weight_type"
          yLabel={[Function]}
        />
        <HealthScoreActions
          date="2017-01-01"
          weights={
            Array [
              Object {
                "weight": 0.5,
                "weight_type": "Hard Bounces",
                "weight_value": 0.25,
              },
              Object {
                "weight": -0.5,
                "weight_type": "Complaints",
                "weight_value": 0.25,
              },
            ]
          }
        />
      </div>
    </Grid.Column>
  </Grid>
  <OtherChartsHeader
    facet="sending-domain"
    facetId="test.com"
  />
  <Grid>
    <Grid.Column
      sm={6}
      xs={12}
    >
      <withRouter(Connect(WithSpamTrapDetails(SpamTrapsPreview))) />
    </Grid.Column>
    <Grid.Column
      sm={6}
      xs={12}
    >
      <withRouter(Connect(withEngagementRecencyDetails(EngagementRecencyPreview))) />
    </Grid.Column>
  </Grid>
</SignalsPage>
`;

exports[`Signals Health Score Page renders empty correctly 1`] = `
<Panel
  sectioned={true}
>
  <ChartHeader
    title="Health Score"
    tooltipContent="
  This is a predictive score that monitors your email health to identify problems before
  they negatively impact email delivery. This score is informed by message events,
  including bounces, spam trap hits, and engagement across our entire network.
"
  />
  <Callout
    title="No Data Available"
  >
    Insufficient data to populate this chart
  </Callout>
</Panel>
`;

exports[`Signals Health Score Page renders empty weights 1`] = `
<SignalsPage
  breadcrumbAction={
    Object {
      "component": [Function],
      "content": "Back to Overview",
      "to": "/signals",
    }
  }
  dimensionPrefix="Health Score for"
  facet="sending-domain"
  facetId="test.com"
  primaryArea={<Connect(DateFilter) />}
>
  <Grid>
    <Grid.Column
      md={7}
      sm={12}
    >
      <Panel
        sectioned={true}
      >
        <ChartHeader
          title="Health Score"
          tooltipContent="
  This is a predictive score that monitors your email health to identify problems before
  they negatively impact email delivery. This score is informed by message events,
  including bounces, spam trap hits, and engagement across our entire network.
"
        />
        <BarChart
          fill="#B157CE"
          gap={0.25}
          height={250}
          margin={
            Object {
              "bottom": 5,
              "left": 18,
              "right": 0,
              "top": 12,
            }
          }
          onClick={[Function]}
          selected="2017-01-01"
          timeSeries={
            Array [
              Object {
                "date": "2017-01-01",
              },
            ]
          }
          tooltipContent={[Function]}
          width="99%"
          xAxisProps={
            Object {
              "tickFormatter": [Function],
              "ticks": Array [],
            }
          }
          xKey="date"
          yAxisProps={
            Object {
              "tickFormatter": [Function],
            }
          }
          yKey="health_score"
          yRange={
            Array [
              "auto",
              "auto",
            ]
          }
        />
        <ChartHeader
          title="Injections"
          tooltipContent="
  Injections are defined as the total number of messages successfully generated or relayed to SparkPost for delivery.
"
        />
        <BarChart
          fill="#B157CE"
          gap={0.25}
          height={190}
          margin={
            Object {
              "bottom": 5,
              "left": 18,
              "right": 0,
              "top": 12,
            }
          }
          onClick={[Function]}
          selected="2017-01-01"
          timeSeries={
            Array [
              Object {
                "date": "2017-01-01",
              },
            ]
          }
          tooltipContent={[Function]}
          width="99%"
          xAxisProps={
            Object {
              "tickFormatter": [Function],
              "ticks": Array [],
            }
          }
          xKey="date"
          yAxisProps={
            Object {
              "tickFormatter": [Function],
            }
          }
          yKey="injections"
          yRange={
            Array [
              "auto",
              "auto",
            ]
          }
        />
      </Panel>
    </Grid.Column>
    <Grid.Column
      md={5}
      mdOffset={0}
      sm={12}
    >
      <div
        className="OffsetCol"
      >
        <ChartHeader
          date="2017-01-01"
          hideLine={true}
          padding="1rem 0 1rem"
          title="Health Score Components"
          tooltipContent="
  Your email Health Score is comprised of several signals, including bounces, spam trap hits, user
  engagement, and others.
"
        />
        <Callout>
          Insufficient data to populate this chart
        </Callout>
        <HealthScoreActions
          date="2017-01-01"
          weights={Array []}
        />
      </div>
    </Grid.Column>
  </Grid>
  <OtherChartsHeader
    facet="sending-domain"
    facetId="test.com"
  />
  <Grid>
    <Grid.Column
      sm={6}
      xs={12}
    >
      <withRouter(Connect(WithSpamTrapDetails(SpamTrapsPreview))) />
    </Grid.Column>
    <Grid.Column
      sm={6}
      xs={12}
    >
      <withRouter(Connect(withEngagementRecencyDetails(EngagementRecencyPreview))) />
    </Grid.Column>
  </Grid>
</SignalsPage>
`;

exports[`Signals Health Score Page renders error correctly 1`] = `
<SignalsPage
  breadcrumbAction={
    Object {
      "component": [Function],
      "content": "Back to Overview",
      "to": "/signals",
    }
  }
  dimensionPrefix="Health Score for"
  facet="sending-domain"
  facetId="test.com"
  primaryArea={<Connect(DateFilter) />}
>
  <Grid>
    <Grid.Column
      md={7}
      sm={12}
    >
      <Panel
        sectioned={true}
      >
        <ChartHeader
          title="Health Score"
          tooltipContent="
  This is a predictive score that monitors your email health to identify problems before
  they negatively impact email delivery. This score is informed by message events,
  including bounces, spam trap hits, and engagement across our entire network.
"
        />
        <Callout
          title="Unable to Load Data"
        >
          error message
        </Callout>
      </Panel>
    </Grid.Column>
    <Grid.Column
      md={5}
      mdOffset={0}
      sm={12}
    >
      <div
        className="OffsetCol"
      >
        <ChartHeader
          date="2017-01-01"
          hideLine={true}
          padding="1rem 0 1rem"
          title="Health Score Components"
          tooltipContent="
  Your email Health Score is comprised of several signals, including bounces, spam trap hits, user
  engagement, and others.
"
        />
      </div>
    </Grid.Column>
  </Grid>
  <OtherChartsHeader
    facet="sending-domain"
    facetId="test.com"
  />
  <Grid>
    <Grid.Column
      sm={6}
      xs={12}
    >
      <withRouter(Connect(WithSpamTrapDetails(SpamTrapsPreview))) />
    </Grid.Column>
    <Grid.Column
      sm={6}
      xs={12}
    >
      <withRouter(Connect(withEngagementRecencyDetails(EngagementRecencyPreview))) />
    </Grid.Column>
  </Grid>
</SignalsPage>
`;

exports[`Signals Health Score Page renders loading correctly 1`] = `
<Panel
  sectioned={true}
>
  <ChartHeader
    title="Health Score"
    tooltipContent="
  This is a predictive score that monitors your email health to identify problems before
  they negatively impact email delivery. This score is informed by message events,
  including bounces, spam trap hits, and engagement across our entire network.
"
  />
  <div
    style={
      Object {
        "height": "220px",
        "position": "relative",
      }
    }
  >
    <Loading />
  </div>
</Panel>
`;
